src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -423,7 +423,7 @@ export class SavedObjectsRepository {
         // When method == 'index' the bulkResponse doesn't include the indexed
         // _source so we return rawMigratedDoc but have to spread the latest
         // _seq_no and _primary_term values from the rawResponse.
-        return this._serializer.rawToSavedObject({
+        return this._rawToSavedObject({
```

Comment:
Don't see anything wrong here, but as it's probably one of the most sensible part of our team's review, could you take a look @rudolf ?

Reply:
I think the previous behaviour here was slightly inconsistent, we returned `namespace` even though most methods didn't. In https://github.com/elastic/kibana/pull/65150 I removed the namespace from being returned in the id, but still missed the `namespace` property. Now all methods will return a `namespaces` property and none a `namespace` property.

It's consistent, but not sure if we want all operations to now return namespaces? Returning it gives the impression that you can edit the value and update it. Makes sense that at least find needs to be able to return namespaces so you can see where the results came from. @legrego @jportner don't know if you've discussed this perhaps?

Reply:
> It's consistent, but not sure if we want all operations to now return namespaces? Returning it gives the impression that you can edit the value and update it. Makes sense that at least find needs to be able to return namespaces so you can see where the results came from. @legrego @jportner don't know if you've discussed this perhaps?

It's not the only field that can't be updated though, so I don't think it adds any more inconsistency. My reasoning for returning `namespaces` everywhere was to have a consistent API, rather than having the return value from `find` differ from the rest. There isn't a technical requirement for this though, so I can change this if you'd prefer.

Reply:
> It's not the only field that can't be updated though

Yes you're right.

If we ignore the fact that there was this inconsistency bug in `find()`, I think we used to treat the `namespace` property as an invisible implementation detail only. The concept of namespaces is visible in the UI, but technically it's always implied through the URL/request all the way down to the scoped saved objects client which finally makes it explicit when it interacts with the repository.

We can "lift" this one level higher and expose it from the repository, or we can expose it all the way to the HTTP API. With `redactSavedObjectsNamespaces` the security plugin seems ready to expose it all the way to the HTTP API.

Although there's a lot in-flight the discussion of exporting the namespace with a saved object for backups, searching across namespaces, and some of the UI mockups for managing multi-namespace objects makes it seem inevitable that this information will need to be returned all the way to the API.

Reply:
>  I think we used to treat the namespace property as an invisible implementation detail only.

You're absolutely right, the namespace property was never intended to be exposed outside of the repository. This was certainly the case when saved objects could only exist within a single space. Now that saved objects an exist in multiple spaces at the same time, we need way a way to surface this to end users. You can imply that the saved object exists in your current space from the URL/request, but it's not possible to see where else it might exist from this context alone.


> We can "lift" this one level higher and expose it from the repository, or we can expose it all the way to the HTTP API. With redactSavedObjectsNamespaces the security plugin seems ready to expose it all the way to the HTTP API.

> Although there's a lot in-flight the discussion of exporting the namespace with a saved object for backups, searching across namespaces, and some of the UI mockups for managing multi-namespace objects makes it seem inevitable that this information will need to be returned all the way to the API.


Yeah, that's the conclusion we've drawn too. We could consider having the spaces plugin transform the `namespaces` array to a `spaceIds` array, and omit it for OSS, but I'm not sure what that would actually buy us.